### ISSUE TITLE
[Feature] java client创建edge创建Edge实例时支持直接指定源和目的vertex的id

### ISSUE THREAD

**Issue by hhhyang** [hugegraph/hugegraph#1483]
## Feature Description:

当前java client创建Edge时需要传入源和目的vertex的实例，但是创建edge时应该只要vertex的id信息。是否支持直接指定vertex的id值，这样可以减少两个vertex实例的创建

当前创建一个edge需要类似这样操作：
Vertex marko = new Vertex("person");
marko.id("marko");
Vertex lop = new Vertex("software");
lop.id("lop");
Edge markoCreateLop = new Edge("created").source(marko).target(lop)
                    .property("date", "2017-12-10").property("weight", 0.8);

是否可以支持以下的方式直接创建：
Edge markoCreateLop = new Edge("created").source("marko").target("lop")
                    .property("date", "2017-12-10").property("weight", 0.8);

**Comment by imbajin** [comment]
client 现有 `sourceId()` 和 `targetId()` 以及 `xxLabel()` 设置的 API , 不过还没支持链式去设置.

你的写法如果只指定 ID, 就缺乏 Label 了, 至少也是 `source(id, label)` 的写法, 跟 Vertex 的 API 改进类似, 会一起考虑的

**Comment by hhhyang** [comment]
@imbajin vertex的id在一个图库中是全局唯一的，跟vertex的label是什么无关。如果直接指定vertex的id的话，应该不需要指定label吧？如果id是primary key类型，那么可以同时指定label和primary key值可以唯一确定一个vertex

**Comment by imbajin** [comment]
是的, 前提如你所说需**要先确定**主键模式, 但是你目前创建边的时候, 没有 label 就无从判断他对应的主键模式, 你的假设是已知了对应 vl 才能的, 所以缺失了这个上文, 目前还不能直接判断是否需要 label.

之后这要改可能把这改成可变参数形式, 然后内部判断不同主键模式, 并保持对其他组件的尽可能兼容
